mlflow/pipelines/steps/ingest/datasets.py

```diff
@@ -583,25 +583,36 @@ class SparkSqlDataset(_SparkDatasetMixin, _Dataset):
     (e.g. `SELECT * FROM my_spark_table`).
     """
 
-    def __init__(self, sql: str, dataset_format: str):
+    def __init__(self, sql: str, location: str, dataset_format: str):
         """
         :param sql: The Spark SQL query string that defines the dataset
                     (e.g. 'SELECT * FROM my_spark_table').
         :param dataset_format: The format of the dataset (e.g. 'csv', 'parquet', ...).
         """
         super().__init__(dataset_format=dataset_format)
         self.sql = sql
+        self.location = location
```

Comment:
`location` only supports `catalog.schema.table` or `schema.table` format, correct? Can we call it `table_path` or `table_location` to be more specific?

Reply:
I think it could read any table. it doesn't need to be of the format mentioned above. Example in the test. We save the table as `spark_df.write.mode("overwrite").saveAsTable("test_table")` and using `spark_session.table(self.location)` we can read the table. So we can support any table. I don't think we need to be restrictive about it unless I am missing something. Thoughts?

I can switch it to `table_location` but I was just trying to be consistent with other variable name associated with `INGEST_DATA_LOCATION`.